pkg/quota/evaluator/core/persistent_volume_claims.go

```diff
@@ -51,68 +54,127 @@ func listPersistentVolumeClaimsByNamespaceFuncUsingClient(kubeClient clientset.I
 	}
 }
 
+// ResourceByStorageClass returns a quota resource name by storage class.
+func ResourceByStorageClass(storageClass string, resourceName api.ResourceName) api.ResourceName {
+	return api.ResourceName(string(storageClass + storageClassSuffix + string(resourceName)))
+}
+
+// V1ResourceByStorageClass returns a quota resource name by storage class.
+func V1ResourceByStorageClass(storageClass string, resourceName v1.ResourceName) v1.ResourceName {
+	return v1.ResourceName(string(storageClass + storageClassSuffix + string(resourceName)))
+}
+
 // NewPersistentVolumeClaimEvaluator returns an evaluator that can evaluate persistent volume claims
 // if the specified shared informer factory is not nil, evaluator may use it to support listing functions.
 func NewPersistentVolumeClaimEvaluator(kubeClient clientset.Interface, f informers.SharedInformerFactory) quota.Evaluator {
-	allResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	fixedResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	usageFunc := PersistentVolumeClaimUsageFunc
+	constraintsFunc := makePersistentVolumeClaimConstraintsFunc(usageFunc)
 	listFuncByNamespace := listPersistentVolumeClaimsByNamespaceFuncUsingClient(kubeClient)
 	if f != nil {
 		listFuncByNamespace = generic.ListResourceUsingInformerFunc(f, schema.GroupResource{Resource: "persistentvolumeclaims"})
 	}
-
 	return &generic.GenericEvaluator{
```

Comment:
I still think the `registry.Store` metaphor is fair.

How about the following:

1. rename `generic.GenericEvalutor` to `generic.Evaluator`
2. implement `generic.SimpleEvaluator` over-top of `generic.Evaluator` to handle plain counts so simple things look simpler
3. allow "other quota" code to still share present and future logic with `generic.Evaluator` akin to `registry.Store`?